packages/core/src/particle/ParticleGenerator.ts

```diff
@@ -261,16 +279,18 @@ export class ParticleGenerator {
           this._resizeInstanceBuffer(false);
         }
       }
-
       emission._emit(lastPlayTime, this._playTime);
-
       if (!main.isLoop && this._playTime > duration) {
         this._isPlaying = false;
       }
     }
 
     // Reset play time when is not playing and no active particles to avoid potential precision problems in GPU
-    if (!this.isAlive) {
+    if (this.isAlive) {
```

Comment:
You separated the logic from the comments. Don’t forget to take away the comments when adjusting the logic.